app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -188,26 +187,38 @@ export default class ComposerEditor extends Component {
 
   @on("didInsertElement")
   _composerEditorInit() {
-    const input = this.element.querySelector(".d-editor-input");
     const preview = this.element.querySelector(".d-editor-preview-wrapper");
+    this._registerImageAltTextButtonClick(preview);
+
+    if (this.composer.allowUpload) {
+      this.uppyComposerUpload.setup(this.element);
+    }
+
+    this.appEvents.trigger(`${this.composerEventPrefix}:will-open`);
+  }
+
+  @bind
+  setupEditor(textManipulation) {
+    this.textManipulation = textManipulation;
+
+    const input = this.element.querySelector(".d-editor-input");
 
     input?.addEventListener(
       "scroll",
       this._throttledSyncEditorAndPreviewScroll
     );
 
-    this._registerImageAltTextButtonClick(preview);
-
     // Focus on the body unless we have a title
     if (!this.get("composer.model.canEditTitle")) {
-      putCursorAtEnd(input);
+      textManipulation.putCursorAtEnd();
```

Comment:
as you have created `this.textManipulation` I would keep using it, avoids issues when you grep after